go/vt/vtorc/logic/topology_recovery.go

```diff
@@ -755,17 +766,42 @@ func CheckAndRecover() {
 		log.Error(err)
 		return
 	}
+
+	// Regardless of if the problem is solved or not we want to monitor active
+	// issues, we use a map of labels and set a counter to `1` for each problem
+	// then we reset any counter that is not present in the current analysis.
+	active := make(map[string][]string)
+	for _, e := range replicationAnalysis {
+		if e.Analysis != inst.NoProblem {
+			names := [...]string{
+				string(e.Analysis),
+				e.AnalyzedInstanceAlias,
+				e.AnalyzedKeyspace,
+				e.AnalyzedShard,
+			}
+
+			key := detectedProblems.GetLabelName(names[:]...)
+			active[key] = names[:]
+			detectedProblems.Set(names[:], 1)
+		}
+	}
+
+	// Reset any non-active problems.
+	for key := range detectedProblems.Counts() {
+		if names, ok := active[key]; !ok {
+			detectedProblems.Set(names, 0)
+		}
```

Comment:
I was looking at the test failures locally and I see now! I missed this while reviewing too!
In this code, if we don't find the `key` in the map, then `names` is going to be empty! So we are calling `detectedProblems.Set(names, 0)` when `names` is nil.

This causes this panic - 
```
goroutine 760 [running]:
vitess.io/vitess/go/stats.(*GaugesWithMultiLabels).Set(0x14000612870?, {0x0?, 0x14000a983c0?, 0x29?}, 0x140007329ee?)
	vitess.io/vitess/go/stats/counters.go:380 +0x80
vitess.io/vitess/go/vt/vtorc/logic.CheckAndRecover()
	vitess.io/vitess/go/vt/vtorc/logic/topology_recovery.go:792 +0x324
vitess.io/vitess/go/vt/vtorc/logic.ContinuousDiscovery.func5.1()
	vitess.io/vitess/go/vt/vtorc/logic/vtorc.go:404 +0xb4
created by vitess.io/vitess/go/vt/vtorc/logic.ContinuousDiscovery.func5 in goroutine 754
	vitess.io/vitess/go/vt/vtorc/logic/vtorc.go:396 +0xd8
```

Reply:
Ah wow nice catch, this is why I've wanted to keep things simple.

Ok I pushed two commits, one adds a `.ResetKey(key)` and another uses it instead of `.Reset(names)`, lmk what you think.

Reply:
This is perfect, this is exactly what I had in mind too!